sdk/include/game-contract-sdk/game_base.hpp

```diff
@@ -320,20 +323,39 @@ class game: public eosio::contract {
 
         sessions.modify(session_itr, get_self(), [&](auto& obj){
             obj.digest = new_digest;
+            obj.last_update = eosio::current_time_point();
         });
 
         on_random(req_id, new_digest);
     }
 
-    GAME_ACTION(init)
-    void init(name platform, name events) {
+    CONTRACT_ACTION(close)
+    void close(uint64_t req_id) {
+        auto session_itr = sessions.require_find(req_id, "session with this req_id not found");
+        eosio::check(is_expired(req_id), "session isn't expired, only expired session can be closed");
+
+        eosio::action(
+            {get_self(),"active"_n},
+            "eosio.token"_n,
+            "transfer"_n,
+            std::make_tuple(get_self(), session_itr->player, session_itr->deposit, std::string("refund"))
```

Comment:
better make transfer  method

`    void transfer(name to, asset quantity) {
        eosio::action(
            eosio::permission_level{_self, "active"_n},
            "eosio.token"_n,
            "transfer"_n,
            std::make_tuple(_self, to, quantity, std::string("player winnings"))
        ).send();
    }`

Reply:
done